src/hostname/hostnamed.c

```diff
@@ -488,6 +492,35 @@ static bool valid_deployment(const char *deployment) {
         return in_charset(deployment, VALID_DEPLOYMENT_CHARS);
 }
 
+static const char* fallback_ansi_color(Context *c) {
+        uint64_t hash;
+        const uint8_t key[16] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+                                   0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f};
+        double h, s = 0.7, v = 0.9;
+        uint8_t r, g, b;
+        const char* hostname;
+        _cleanup_free_ char *color = NULL;
+        int k;
+        struct siphash state;
+
+        assert(c);
+
+        hostname = isempty(c->data[PROP_STATIC_HOSTNAME_SUBSTITUTED_WILDCARDS]) ? \
+                "localhost" : c->data[PROP_STATIC_HOSTNAME_SUBSTITUTED_WILDCARDS];
+        siphash24_init(&state, key);
+        string_hash_func(hostname, &state);
+        hash = siphash24_finalize(&state);
+        h = (double)(hash % 360);
+        hsv_to_rgb(h, s, v, &r, &g, &b);
+        k = asprintf(&color, "38;2;%u;%u;%u", r, g, b);
+        if (k < 0) {
+                log_oom();
+                return NULL;
+        }
+
+        return color;
```

Comment:
you need to use TAKE_PTR() here, otherwise this is a memory corruption because the `_cleanup_free_` handler on the variable will still hit.